packages/Kokkos/src/DTK_PolyhedronList.hpp

```diff
@@ -0,0 +1,156 @@
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \file DTK_PolyhedronList.hpp
+ * \brief Polyhedron list.
+ */
+//---------------------------------------------------------------------------//
+
+#ifndef DTK_POLYHEDRONLIST_HPP
+#define DTK_POLYHEDRONLIST_HPP
+
+#include "DTK_ConfigDefs.hpp"
+
+#include <Kokkos_Core.hpp>
+
+namespace DataTransferKit
+{
+
+//---------------------------------------------------------------------------//
+/*!
+ * \class PolyhedronList.
+ *
+ * \brief Trivially-copyable polyhedron list.
+ *
+ * \tparam ViewProperties Properties of the contained Kokkos views.
+ *
+ * PolyhedronList describes arbitrary linear polyhedra accessible from the
+ * calling MPI rank.
+*/
+template <class... ViewProperties>
+class PolyhedronList
+{
+  public:
+    //! View traits.
+    using ViewTraits = typename Kokkos::ViewTraits<int, ViewProperties...>;
+
+    //! The coordinates of the nodes that build the polyhedrons that are
+    //! provided through the cell list on this MPI rank. This view is rank-2
+    //! and should be sized as (number of nodes, spatial dimension).
+    Kokkos::View<Coordinate **, ViewProperties...> coordinates;
+
+    //! Connectivity list of faces that build the polyhedrons that are local
+    //! to this MPI rank. This view is defined as rank-1 but represents
+    //! unstructured rank-2 data. It should be sized as (total sum of the
+    //! number of nodes composing each face) or the sum of all elements in the
+    //! following view, nodes_per_face, which indicates how many nodes are
+    //! assigned to each face and how to index into this view. The input
+    //! should be arranged as follows. Consider the \f$n^th\f$ node of face
+    //! \f$i\f$ to be \f$f^i_n\f$ which is equal to the local index of the
+    //! corresponding node in the coordinates view. Two faces, the first with
+    //! 4 nodes and the second with 3 would then be defined via this view as:
+    //! \f$(f^1_1, f^1_2, f^1_3, f^1_4, f^2_1, f^2_2, f^2_3 )\f$ with the
+    //! nodes_per_face view reading \f$(4, 3)\f$
+    Kokkos::View<LocalOrdinal *, ViewProperties...> faces;
+
+    //! Number of nodes composing each face in the face input view. The sum of
+    //! all local elements in this view should equal the total size of the
+    //! faces view. This view rank-1 and should be of length of the number of
+    //! faces in the list.
+    Kokkos::View<int *, ViewProperties...> nodes_per_face;
```

Comment:
Should we make all integers unsigned? I can't really think of any of the data structures that need signed integers.